src/mame/drivers/dfs500.cpp

```diff
@@ -0,0 +1,1036 @@
+// license:GPL-2.0+
+// copyright-holders:Felipe Sanches
+/****************************************************************************
+
+    Sony DFS-500 DME Video Mixer
+
+    An artwork layout for this is under development at:
+    https://github.com/felipesanches/dfs500_MAME_artwork
+
+    ===== Current status of this driver =====
+    Upon power up, the ROM version number is displayed: "1.03"
+    And after a while, we get an error: "Er02"
+
+    This is likely due to bugs in this driver rendering the control panel unable to
+    communicate properly with the video mixer maincpu via a serial interface.
+
+    Apparently the maincpu does not respond to the control panel commands because it gets
+    lost earlier, while trying to communicate with the secondary cpu (the "effects" cpu)
+    via a pair of 16 bit latches.
+
+    The effects CPU does not yet reply due to some other problem in the driver.
+    I guess it is related to the "OREG2" in its memory map which have not yet been
+    properly declared. We'll need to further study that portion of the circuit in the
+    service manual in order to finish declaring the effects_cpu memory map layout.
+
+    ===== Usage: =====
+    According to the service manual, the LED test mode can be enabled by simultaneously
+    pressing these buttons in the control panel:
+    * FOREGROUND VTR A
+    * BACKGROUND VTR B
+    * Location
+    With the default key mappings in this driver, this can be achieved by pressing Q + 2 + L
+
+    * All LEDs and 7-seg displays blink sequentialy.
+    * If you press any button on the control panel, the sequence of blinking resumes from
+      that selected area of the panel.
+    * To stop the blinking sequence you can press the keypad ENTER key.
+
+    ===== Notes on the video hardware: =====
+    This video mixer accepts 4 input video signals to be mixed. This driver currently emulates
+    this by using static PNG images provided in the command line with the -inputN flags (with N = 1 to 4)
+
+    The DSP circuitry was not yet emulated, and the minimal video code in this driver currently
+    simply bypasses the input #1 into the output. Once the serial communication between control panel and
+    maincpu is fixed, we should at least see the video output changing when pressing the BACKGROUND input
+    selector buttons VTR A, VTR B, 3 and 4 (to select ammong the 4 video inputs).
+
+****************************************************************************/
+#define VIDEO_WIDTH 768
+#define VIDEO_HEIGHT 256
```

Comment:
Don’t put anything in a compilation unit besides comments before you include the prefix header.

Reply:
what is "the prefix header"?

Reply:
I think he means the include of emu.h

Reply:
if that's the case, then now we're good :-)

Reply:
emu.h is pre-compiled and supplied as a prefix header in the default configuration.